Houseclub/src/main/java/me/grishka/houseclub/fragments/ProfileFragment.java

```diff
@@ -295,4 +319,40 @@ public void onError(ErrorResponse error){
 				.setNegativeButton(R.string.cancel, null)
 				.show();
 	}
+
+	@Override
+	public void onActivityResult(int requestCode, int resultCode, Intent data)
+	{
+		if (requestCode == ProfileFragment.IMAGE_PICKER) {
+			if (data == null) {
+				return;
+			}
+			Uri u = data.getData();
+			updateAvatar(u);
+		}
+	}
+
+
+	public void setAvatar(Uri uri) {
+		photo.setImageURI(uri);
+	}
+
+	public void updateAvatar(Uri file) {
+		new UpdatePhoto(file)
+				.wrapProgress(getActivity())
+				.setCallback(new SimpleCallback<BaseResponse>(this){
+					@Override
+					public void onSuccess(BaseResponse result){
+						if (!result.success) {
```

Comment:
Failed requests trigger the onError callback, and `ErrorResponse` also has the `showToast()` method to do just this.